[DNM] tpcc: remove ORDER BY in newOrder txn

petermattis opened the pull request.

petermattis committed:
[DNM] tpcc: remove ORDER BY in newOrder txn

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/loadgen/159)
<!-- Reviewable:end -->

petermattis commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[tpcc/new_order.go, line 284 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/159#-L70ndPby2rAToyMOkXY:-L70ndPby2rAToyMOkXZ:b-nr47m) ([raw file](https://github.com/cockroachdb/loadgen/blob/7638eeb35ea98a71c8502d4b66bb36def63516a4/tpcc/new_order.go#L284)):*
> ```Go
> 
> 				var j int
> 				for ; j < len(itemIDs); j++ {
> ```

Could binary search `itemIDs` here, but this seemed more obvious. Note that `itemIDs` does not contain duplicates.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/159)*
<!-- Sent from Reviewable.io -->

nvb commented on tpcc/new_order.go:
```diff
@@ -272,10 +280,25 @@ func (n newOrder) run(db *sql.DB, wID int) (interface{}, error) {
+
+				ids[j] = sID
+				distInfos[j] = sDistInfo
+				sQuantityUpdateCases[j] = fmt.Sprintf("WHEN %s THEN %d", stockIDs[i], newSQuantity)
```
`s/stockIDs[i]/stockIDs[j]/` in each of these lines, right?

petermattis commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[tpcc/new_order.go, line 298 at r1](https://reviewable.io:443/reviews/cockroachdb/loadgen/159#-L70nBB1UpRH0_fkN3-J-r1-298:-L70nyKCi50o6SD45cWG:b-vkmc62) ([raw file](https://github.com/cockroachdb/loadgen/blob/7638eeb35ea98a71c8502d4b66bb36def63516a4/tpcc/new_order.go#L298)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

`s/stockIDs[i]/stockIDs[j]/` in each of these lines, right?
</blockquote></details>

Oh, shoot. Yes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/159)*
<!-- Sent from Reviewable.io -->

nvb commented on tpcc/new_order.go:
```diff
@@ -245,12 +248,17 @@ func (n newOrder) run(db *sql.DB, wID int) (interface{}, error) {
 				item := &d.items[i]
 
 				if !rows.Next() {
+					fmt.Printf(`%d: missing stock row: retry=%d %s
```
I'm having a hard time reasoning about this. I think it would be easier if we made the following changes:
- get rid of the `i` index variable
- determine `j` immediately after the `rows.Scan` call
- move `item := &d.items[i]` down and change it to `item := &d.items[j]`

petermattis commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[tpcc/new_order.go, line 251 at r2](https://reviewable.io:443/reviews/cockroachdb/loadgen/159#-L70nBB1UpRH0_fkN3-J-r2-251:-L70q3_S7ZjjYrDGCPWO:b-3tw0rf) ([raw file](https://github.com/cockroachdb/loadgen/blob/3dd99516e447e83bcc2848296348bd40702aa9a2/tpcc/new_order.go#L251)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

I'm having a hard time reasoning about this. I think it would be easier if we made the following changes:
- get rid of the `i` index variable
- determine `j` immediately after the `rows.Scan` call
- move `item := &d.items[i]` down and change it to `item := &d.items[j]`
</blockquote></details>

I renamed the loop variable from `i` to `k`, renamed `j` to `i` and took the other suggestions. I don't think this actually changed the correctness of anything other than setting `item.brandGeneric`. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/159)*
<!-- Sent from Reviewable.io -->

nvb commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[tpcc/new_order.go, line 251 at r2](https://reviewable.io:443/reviews/cockroachdb/loadgen/159#-L70nBB1UpRH0_fkN3-J-r2-251:-L70qdZFI64Qv9P8gdJG:b-9y3ulu) ([raw file](https://github.com/cockroachdb/loadgen/blob/3dd99516e447e83bcc2848296348bd40702aa9a2/tpcc/new_order.go#L251)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I renamed the loop variable from `i` to `k`, renamed `j` to `i` and took the other suggestions. I don't think this actually changed the correctness of anything other than setting `item.brandGeneric`. 
</blockquote></details>

Yeah, you might be right. It's a lot easier to understand now though.

It looks good to me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/loadgen/159)*
<!-- Sent from Reviewable.io -->

petermattis committed:
[WIP] tpcc: removing tracing

petermattis committed:
[WIP] trim 1

petermattis committed:
[WIP] trim 2

petermattis committed:
[WIP] trim 3

petermattis committed:
[WIP] trim 4

petermattis committed:
[WIP] trim 5

petermattis committed:
[WIP] trim 6

petermattis committed:
[WIP] trim 7

petermattis committed:
[WIP] trim 8

petermattis committed:
[WIP] trim 9

petermattis committed:
[WIP] trim 10
